File: app/src/main/java/com/futsch1/medtimer/database/JSONReminderEventBackup.java
package com.futsch1.medtimer.database;

import com.google.gson.GsonBuilder;

import java.util.List;

public class JSONReminderEventBackup extends JSONBackup<ReminderEvent> {

    public JSONReminderEventBackup() {
        super(ReminderEvent.class);
    }

    @Override
    protected GsonBuilder registerTypeAdapters(GsonBuilder builder) {
        return builder
                .registerTypeAdapter(ReminderEvent.class, new FullDeserialize<ReminderEvent>());
    }

    protected boolean isInvalid(ReminderEvent reminderEvent) {
        return reminderEvent == null;
    }

    public void applyBackup(List<ReminderEvent> listOfReminderEvents, MedicineRepository medicineRepository) {
        medicineRepository.deleteReminderEvents();

        listOfReminderEvents.forEach(medicineRepository::insertReminderEvent);
    }
}

File: app/src/main/java/com/futsch1/medtimer/database/JSONReminderEventBackup.kt
package com.futsch1.medtimer.database

import com.google.gson.GsonBuilder

class JSONReminderEventBackup(
    private val reminderEventDao: ReminderEventDao
) : JSONBackup<ReminderEventEntity>(ReminderEventEntity::class.java) {
    override fun isInvalid(item: ReminderEventEntity?): Boolean {
        return item == null
    }

    override fun registerTypeAdapters(builder: GsonBuilder): GsonBuilder {
        return builder
            .registerTypeAdapter(ReminderEventEntity::class.java, FullDeserialize<ReminderEventEntity>())
    }

    override suspend fun applyBackup(list: List<ReminderEventEntity>) {
        reminderEventDao.deleteAll()
        reminderEventDao.createAll(list)
    }
}
